src/jit/lsraxarch.cpp

```diff
@@ -2297,224 +2297,253 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
+                assert(!isRMW);
+                assert(dstCount == 1);
+
+                if (!op1->isContained())
                 {
-                    buildInternalFloatRegisterDefForNode(intrinsicTree, allSIMDRegs());
+                    tgtPrefUse = BuildUse(op1);
+                    srcCount   = 1;
+                }
+                else
+                {
+                    srcCount += BuildOperandUses(op1);
                 }
+
+                buildUses = false;
+                break;
             }
-            break;
-        }
 
-        case NI_SSE2_MaskMove:
-        {
-            // SSE2 MaskMove hardcodes the destination (op3) in DI/EDI/RDI
-            assert(!intrinsicTree->isRMWHWIntrinsic(compiler));
-            // MaskMove doesn't have RMW semantics.
-            assert(!isRMW);
-            BuildUse(op1);
-            BuildUse(op2);
-            BuildUse(op3, RBM_EDI);
-            buildUses = false;
-            break;
-        }
+            case NI_AVX_SetAllVector256:
+            {
+                if (varTypeIsIntegral(baseType))
+                {
+                    buildInternalFloatRegisterDefForNode(intrinsicTree, allSIMDRegs());
+                    if (!compiler->compSupports(InstructionSet_AVX2) && varTypeIsByte(baseType))
+                    {
+                        buildInternalFloatRegisterDefForNode(intrinsicTree, allSIMDRegs());
+                    }
+                }
+                break;
+            }
 
-        case NI_SSE41_BlendVariable:
-            if (!compiler->canUseVexEncoding())
+            case NI_SSE2_MaskMove:
             {
                 assert(numArgs == 3);
-                assert(intrinsicTree->isRMWHWIntrinsic(compiler));
-                assert(isRMW);
-                // SSE4.1 blendv* hardcode the mask vector (op3) in XMM0
-                BuildUse(op1);
-                RefPosition* op2Use = BuildUse(op2);
-                setDelayFree(op2Use);
-                RefPosition* op3Use = BuildUse(op3, RBM_XMM0);
-                setDelayFree(op3Use);
+                assert(!isRMW);
+
+                // MaskMove hardcodes the destination (op3) in DI/EDI/RDI
+                srcCount += BuildOperandUses(op1);
+                srcCount += BuildOperandUses(op2);
+                srcCount += BuildOperandUses(op3, RBM_EDI);
+
                 buildUses = false;
-                srcCount  = 3;
+                break;
             }
-            break;
-
-        case NI_SSE41_TestAllOnes:
-        {
-            buildInternalFloatRegisterDefForNode(intrinsicTree);
-            break;
-        }
 
-        case NI_SSE41_Extract:
-            if (baseType == TYP_FLOAT)
+            case NI_SSE41_BlendVariable:
             {
-                buildInternalIntRegisterDefForNode(intrinsicTree);
+                assert(numArgs == 3);
+
+                if (!compiler->canUseVexEncoding())
+                {
+                    assert(isRMW);
```

Comment:
Moved the assert into the if statement, since it is only RMW on non-VEX hardware.